### ISSUE TITLE
Support Python 3.4.1 runtime?

### ISSUE THREAD

**Issue by wmayner** [thenovices/heroku-buildpack-scipy#14]
Thanks for this very useful buildpack! Is there any way to get it working with 3.4?

**Comment by abezzub** [comment]
I also need Python 3.4 support. 

@thenovices can you please explain how you build packages? I see that https://github.com/thenovices/npscipy-binaries mentions using virtual machines, do you have some scripts to do it or do you do everything manually?

**Comment by thenovices** [comment]
Sorry, I'm pretty busy for the next two months so probably won't get to it... I build the packages manually. Basically, the amount of repetition thus far has been too low to merit the investment needed to script it...

**Comment by abezzub** [comment]
No problem. Do you just use VM with Ubuntu of the same version that Heroku uses? Or are there some additional requirements? Thanks

**Comment by thenovices** [comment]
Yeah, I think the three things are:
1. Correct version of Ubuntu
2. Correct version of GCC
3. Same Python. In the case of Python 2, this meant that the Python had to be compiled with UCS2 instead of UCS4 (iirc). Not sure if this same logic applies for Python 3. But if I compiled the numpy package with a different Python (i.e., compiled with UCS4), then I would get a runtime error when I ran numpy on Heroku.

**Comment by andrewychoi** [comment]
Quick note: for the new `cedar-14` stack, `sys.maxunicode` is 1114111, which would indicate Python compiled against UCS-4

**Comment by thenovices** [comment]
Thanks, good to know!

**Comment by andrewychoi** [comment]
Revision to the above: the Python version on the new `cedar-14` stack is actually 2.7.9, and `sys.maxunicode` is 65535. I think what happened was that I ran the code on the old `cedar` stack, and on that stack, the default Python version is 2.7.6 and `sys.maxunicode` is 1114111.

In any case, I've built a Vagrant script and a few shell scripts to interact with the VM. Hopefully this bridges some of the gap to build new versions.

The scripts are here: https://github.com/andrewychoi/npscipy-binaries and more specifically: https://github.com/andrewychoi/npscipy-binaries/tree/master/vagrant

**Comment by feliperyan** [comment]
@andrewychoi did you manage to get a python 3.4 version of this going? Anyone else?

**Comment by abezzub** [comment]
I ended up using conda to install math packages. There was a Python 2 buildpack for conda that I tweaked to work with Python 3: https://github.com/teamupstart/conda-buildpack

**Comment by andrewychoi** [comment]
@feliperyan No, I haven't tried getting this working on py3.4, but I'd imagine the changes are straightforward to make in the vagrant provisioning file: https://github.com/andrewychoi/npscipy-binaries/blob/master/vagrant/provision_numpy_scipy.sh

**Comment by feliperyan** [comment]
@andrewychoi Thanks for this Andrew, I'm new to Vagrant - I can see the Vagrantfile picks up on provision_numpy_scipy.sh but there are a number of other bash scripts in there. They seem to be auxiliary scripts to be ran by the host which do a few things in the box. 

Do I need to worry about them at all? They make references to python 2.7

**Comment by andrewychoi** [comment]
Yes, they're scripts I used to construct the binary packaging. Essentially they ssh into the vagrant image, and copy the associated numpy/scipy files out to a new zipfile.

Looking at a couple of py3.4 projects I have, you'll need to change the prefix from `venv/lib/python2.7` to `venv/lib/python3.4` in those scripts.

**Comment by feliperyan** [comment]
Thank you @andrewychoi I might still give this a go at a later stage but for now I have followed @abezzub and got it working with the condo buildpack he kindly shared with us. 
Btw my intention was to get Palladium working on Heroku, which I now have done.

**Comment by FreshDonut** [comment]
Any plans to still support 3.4 or is that no longer happening? Thanks! :smiley:

**Comment by branliu0** [comment]
I'll get around to it someday, but I'm pretty busy these days :/

**Comment by FreshDonut** [comment]
No worries. : )

**Comment by feliperyan** [comment]
Yo,
I ended up getting it to work. My goal was to run palladium on it. From
memory it involved using P3.4

**Comment by brianthomas** [comment]
I've forked this project to support additional libraries (gensim) for textmining under python 3.4.x. You can pull the binaries from that project which should make it easy to upgrade this one to 3.4.x. See : 

https://github.com/brianthomas/ubuntu-textmining-binaries

which support the textmining buildpack:

https://github.com/brianthomas/heroku-buildpack-textmining
